file_formats/stimulus_tables.md

```diff
@@ -0,0 +1,118 @@
+
+### File format
+
+Stim tables may be produced from any source format. In the aind-metadata-mapper, there are various ETL objects each associated with different projects and rig types. These ETLs should identify the correct source file containing stimulus info and generate a proper stim table in the form of a .CSV. For instance, the `CamstimEphysSessionEtl` ETL uses either a `*.stim.pkl` or a `*.behavior.pkl` depending on which is present, and uses an `*.opto.pkl` if it is present. The ETL is responsible for handling the variations between the structure of these files. In this case the directory structure may look as follows.
+
+e.g.:
+
+```plaintext
+
+📦np-exp (or other session directory)
+┣ 📂<session_id> (e.g 0123456789_123456_20250101)
+┃  ┗ 📜<session_id>.behavior.pkl
+┃  ┗ 📜<session_id>.opto.pkl
+...
+```
+
+### Application notes
+
+TODO
+This section is reserved to provide additional information on how to acquire the data in the data format described above. It can include information relative to the hardware (e.g. supported models), software interface (e.g. SoftwareFoo with version >= 1.0.0), ideally, some easy to deploy or follow examples that can get anyone to reproduce the data format.
+
+
+
+### Relationship to aind-data-schema
+
+There is no relationship between the raw data and the aind-data-schema that is relevant here. The raw stimulus information relates to the aind-data-schema through the derived asset, which is described below.
+
+
+### File Quality Assurances
+
+TODO
+This section is reserved to describe what features of the data format should be true if the data asset is to be considered valid. Conceptually, this section should describe features that can be easily tested and validated by unit tests. Examples include:
+- "There will always be two files: `data.dat` and `metadata.txt`"
+- "For each frame in `video.avi`, there will be a corresponding row in `metadata.csv`"
+- "Field `Bar` will always be a positive integer"
+- "The first timestamp value in `metadata_camera.csv` will always be greater than the first one in `metadata_behavior.csv`
+- The `Time` column in `file.bin` is assumed to be aligned (sharing the same time domain) with `Time` column of `another_file.csv`
+
+
+## Primary Data Format
+
+### File format
+
+#### Basic Stimulus Table
+The stim tables are stored as `.CSV` tables. At minimum, it must include the following columns:
+- `start_time`: float representing the time, in seconds, the stimulus began
+- `stop_time`: float representing the time, in seconds, the stimulus ended
+- `stim_name`: string representing the unique kind of stim shown. E.g "receptive_field_mapping", "drifting_gratings", "natural_movie_1"
+
+Any stimulus which is not procedurally generated should have additional columns. 
+For image sequences:
+- `image_name`: string representing the name of the image. This is used to associate the stimulus with a stim_template object/file which should have the same name in subsequent data packaging.
```

Comment:
We should consider whether a string makes sense or an index if more appropriate.